src/v/config/configuration.cc

```diff
@@ -144,6 +148,26 @@ configuration::configuration()
         .max = 1000 // A system with 1M ulimit should be allowed to create at
                     // least 1000 partitions
       })
+  , topic_partitions_per_shard(
+      *this,
+      "topic_partitions_per_shard",
+      "Maximum number of partitions which may be allocated to one shard (CPU "
+      "core)",
+      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      7000,
+      {.min = 16, // Forbid absurdly small values that would prevent most
+                  // practical workloads from running
+       .max = static_cast<uint32_t>(std::numeric_limits<int32_t>::max())})
```

Comment:
why this kind of max value, is there a reasoning behind it ?

Reply:
This was kind of a compromise. I'm using an unsigned because the value can't reasonably be negative, but it's also convenient to do some signed arithmetic when updating the state of allocation_node, so I want the value within the signed int bounds.